core/src/services/routing.js

```diff
@@ -187,35 +187,41 @@ export const handleRouteChange = async (path, component, node, config) => {
 
     if (!viewUrl) {
       const routeExists = RoutingHelpers.isExistingRoute(pathUrl, pathData);
-
+      const defaultChildNode = await RoutingHelpers.getDefaultChildNode(
+        pathData
+      );
       if (routeExists) {
-        const defaultChildNode = await RoutingHelpers.getDefaultChildNode(
-          pathData
-        );
+        //normal navigation can be performed
         navigateTo(`${pathUrl ? `/${pathUrl}` : ''}/${defaultChildNode}`);
       } else {
-        const alert = {
-          message: 'Could not find the requested route',
-          link: pathUrlRaw
-        };
-
-        component.set({ alert });
-        navigateTo('/');
+        if (defaultChildNode && pathData.navigationPath.length > 2) {
```

Comment:
1. In case (probably impossible) `pathData.navitationPath` was empty, there will be no errors.
2. I've just changed it to `1` because `overview` is a defaultChild for `/`. If it was set to _>0_ or not present at all, "Could not find exact route" error would appear instead of "Could not find requested route" due to automatic redirection to `/`. 

Thanks for noticing `2` was the wrong number 👍

Reply:
I would rather create a variable out of `pathData.navigationPath.length > 1` to make clear what are we checking here through the variable name.